File: jpress-search/src/main/java/by.itx.jpress.plugins.search/SearchPlugin.java
package by.itx.jpress.plugins.search;

import by.itx.jpress.api.Plugin;

/**
 * Base class for implementing by plugin developers.
 *
 * @author Ibragimov Ruslan
 */
public class SearchPlugin extends Plugin {


    public SearchPlugin() {
        super("search");
        System.out.println("Search!");
    }


}

File: kpress-plugins/kpress-search/src/main/kotlin/io/heapy/kpress/plugins/search/SearchPlugin.kt
package io.heapy.kpress.plugins.search

import io.heapy.kpress.plugin.api.Plugin
import java.util.EventListener

/**
 * Search plugin
 *
 * @author Ruslan Ibragimov
 * @since 0.1
 */
class SearchPlugin : Plugin {

    override val namespace: String = "search"

    override fun addEventListener(listener: EventListener) {

    }
}
